File: payroll-api/src/main/java/com/company/payroll/controller/api/StaffSalaryController.java
package com.company.payroll.controller.api;

import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import com.company.payroll.model.StaffSalary;
import com.company.payroll.service.StaffSalaryService;
import com.github.pagehelper.PageInfo;

import io.swagger.v3.oas.annotations.Operation;

@RestController
@RequestMapping("/api/staff/salary")
public class StaffSalaryController {
	
	@Autowired
	private StaffSalaryService staffSalaryService;
	
	@Operation(summary="Get salary list")
	@GetMapping
	public ResponseEntity<PageInfo<StaffSalary>> list(@RequestParam(value="page", required=true) int page,
													  @RequestParam(value="size", required=true) int offset) {
		return ResponseEntity.ok(staffSalaryService.list(page, offset));
	}
	
	@Operation(summary="Get salary info by id")
	@GetMapping("/{id}")
	public ResponseEntity<Optional<StaffSalary>> findById(@PathVariable("id") Integer sId) {
		return ResponseEntity.ok(staffSalaryService.findById(sId));
	}
	
	@Operation(summary="Update staffSalary info.")
	@PutMapping("/{id}")
	public ResponseEntity<StaffSalary> update(@RequestBody StaffSalary staffSalary) {
		return ResponseEntity.ok(staffSalaryService.update(staffSalary));
	}
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/controller/api/StaffSalaryController.kt
package com.company.payroll.controller.api

import com.company.payroll.model.StaffSalary
import com.company.payroll.service.StaffSalaryService
import com.github.pagehelper.PageInfo
import io.swagger.v3.oas.annotations.Operation
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/staff/salary")
class StaffSalaryController(@Autowired private val staffSalaryService: StaffSalaryService) {
  @Operation(summary = "Get salary list")
  @GetMapping
  fun listSalary(@RequestParam(value = "page", required = true) page: Int, @RequestParam(value = "size", required = true) offset: Int): ResponseEntity<PageInfo<StaffSalary>> {
    return ResponseEntity.ok(staffSalaryService.list(page, offset))
  }

  @Operation(summary = "Get salary info by id")
  @GetMapping("/{id}")
  fun getById(@PathVariable("id") sid: Int): ResponseEntity<StaffSalary?> {
    return ResponseEntity.ok(staffSalaryService.findById(sid))
  }

  @Operation(summary = "Update salary info.")
  @PutMapping("/{id}")
  fun update(@RequestBody staffSalary: StaffSalary): ResponseEntity<StaffSalary> {
    return ResponseEntity.ok(staffSalaryService.update(staffSalary))
  }
}